### PR TITLE
# 165020073 Setup end to end and units tests

### PR DESCRIPTION
**Pull request by giseleiradu**
#### What does this PR do?

- Sets up Jest as a testing engine for unit testing

- Sets up a Cypress as an end to end testing library

#### Description of Task to be completed?

- Installing Jest and Cypress

- Setting their command in the package.json

#### How should this be manually tested?

-Jest: ```npm test``` 

- Cypress: ```npm start``` runs first this Cypress before displaying the page or ```npm run cypress``` to just open Cypress

#### What are the relevant pivotal tracker stories?
[#165020073](https://www.pivotaltracker.com/story/show/165020073)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: setupTests.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// import {configure} from 'enzyme';
```
Comment (Karlmusingo) [inline setupTests.js]: Why is this file commented? I suggest to delete it if it's not used anymore.

File: cypress.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "baseUrl": "http://localhost:3000"
```
Comment (dusmel) [inline cypress.json:3]: Is there a way to make this port dynamic since our  webpack-dev-server does it manually...?

**Review by Karlmusingo — COMMENTED** [review]
@giseleiradu, I see two test folders in your branch, I suggest to use one(__test__)

**Review by dusmel — CHANGES_REQUESTED** [review]
I think for Hound you should fix the errors due to semicolons and regarding the es6 support it should be fixed when we have eslint config from #5

File: setupTests.js (hunk 1 above)
Comment (giseleiradu) [reply setupTests.js]: Thank you @Karlmusingo, I am on it.

File: cypress.json (hunk 2 above)
Comment (giseleiradu) [reply cypress.json:3]: Up to now, I don't see how. But I'll keep that in mind for near implementation

Comment (dusmel) [reply cypress.json:3]: cooolio 👍

**Review by dusmel — APPROVED** [review]

**Pull request merged**